feg/cloud/go/services/feg_relay/utils/get_all_gateways.go

```diff
@@ -56,31 +56,35 @@ func GetAllGatewayIDs(ctx context.Context) ([]string, error) {
 	}
 	// Find as many gateways as possible, don't exit on error, just return last error to the caller along with
 	// the list of GWs found
+	var retErr error
 	for _, networkID := range cfg.ServedNetworkIds {
 		gateways, _, err := configurator.LoadEntities(
 			networkID, swag.String(orc8r.MagmadGatewayType), nil, nil, nil,
 			configurator.EntityLoadCriteria{},
 			serdes.Entity,
 		)
 		if err != nil {
-			glog.Errorf("List Network '%s' Gateways error: %v", networkID, err)
+			retErr = fmt.Errorf("List Network '%s' Gateways error: %v", networkID, err)
+			glog.Errorf("%+v", retErr)
 			continue
 		}
 		for _, gatewayEntity := range gateways {
 			record, err := device.GetDevice(networkID, orc8r.AccessGatewayRecordType, gatewayEntity.PhysicalID, serdes.Device)
 			if err != nil {
-				err = fmt.Errorf("Find Gateway Record Error: %v for Gateway %s:%s", err, networkID, gatewayEntity.Key)
+				retErr = fmt.Errorf("Find Gateway Record Error: %v for Gateway %s:%s", err, networkID, gatewayEntity.Key)
+				glog.Errorf("%+v", retErr)
 				continue
 			}
 			hardwareID := record.(*orc8rModels.GatewayDevice).HardwareID
 			if len(hardwareID) > 0 {
 				res = append(res, hardwareID)
 			} else {
-				err = fmt.Errorf("Empty Harware ID for Gateway %s:%s", networkID, gatewayEntity.Key)
+				retErr = fmt.Errorf("Empty Harware ID for Gateway %s:%s", networkID, gatewayEntity.Key)
+				glog.Errorf("%+v", retErr)
 			}
 		}
 	}
-	return res, err
+	return res, retErr
```

Comment:
You are changing logic here, if anything, I'd relax failure conditions.
Check out the comments on the function use cases: "// Reset ALL is a "best effort" request, we reset what we can & just log error for what we cannot"

GetAllGatewayIDs() is used when we need opportunistically notify all network gateways of something, so - notifying some is better than none.

Just do:
if len(res) > 0 {
    retErr = nil
}
return res, retErr

If you want to keep the changes to this file

Reply:
The issue with this snippet is the error was never actually getting propagated, because the outer `err` is being shadowed by the inner `err`. I've amended the change slightly to follow the same logic but properly handle the shadowing, let me know if you're content with the change from here

Reply:
> The issue with this snippet is the error was never actually getting propagated, because the outer `err` is being shadowed by the inner `err`...
Exactly - the point of this function should NOT to return error if even a single GW ID is found (logging all failures instead).
You can change it to return an intermediate error if there are zerow GWs found anyway, but please return error nil if there is even a single valid GW in the result list.

So, the suggested 

if len(res) > 0 {
    retErr = nil
}
return res, retErr

should do it 4 you.

Reply:
From offline: amending to follow the actual (rather than stated) behavior of the function. Will pass to @uri200 in a separate PR if he wants to change anything

Reply:
Per offline discussion, it looks fine.  In case of `partial success` do not error and go ahead.

The only draw back is the partial error log will only show in orc8r. But your implementation is in line with other implementations in the system and it is less confusing .